scripts/artifactory-chart-releaser/main.py

```diff
@@ -0,0 +1,64 @@
+import os
+import subprocess
+import yaml
+
+ARTIFACTORY_REPOSITORY_URL = os.getenv("ARTIFACTORY_REPOSITORY_URL")
+
+def get_chart_info(chart_dir):
+    """Extract chart name and version from Chart.yaml"""
+    with open(os.path.join(chart_dir, "Chart.yaml"), "r") as file:
+        chart_yaml = yaml.safe_load(file)
+        chart_name = chart_yaml.get("name")
+        chart_version = chart_yaml.get("version")
+    return chart_name, chart_version
+
+def helm_chart_exists(chart_name, chart_version):
+    """Check if a Helm chart exists in the repository"""
+    try:
+        subprocess.run(
+            ["helm", "pull", f"oci://{ARTIFACTORY_REPOSITORY_URL}/{chart_name}", "--version", chart_version],
+            check=True,
+            capture_output=True,
+            text=True
+        )
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+def upload_chart(chart_dir):
+    """Upload Helm chart to OCI repository"""
+    chart_name, chart_version = get_chart_info(chart_dir)
+    print(f"Uploading chart {chart_name} with version {chart_version}")
+
+    if helm_chart_exists(chart_name, chart_version):
+        print(f"Helm chart {chart_name} with version {chart_version} already exists in {ARTIFACTORY_REPOSITORY_URL}. Skipping...")
+        return
+
+    # Update dependencies, package the chart, and push to the repository
+    subprocess.run(["helm", "dependency", "update"], check=True, cwd=chart_dir)
+    subprocess.run(["helm", "package", "."], check=True, cwd=chart_dir)
+    
+    package_file = f"{chart_name}-{chart_version}.tgz"
+    try:
+        subprocess.run(
+            ["helm", "push", package_file, f"oci://{ARTIFACTORY_REPOSITORY_URL}"],
+            check=True,
+            cwd=chart_dir
+        )
+        print(f"Pushed helm chart {package_file} to the OCI repository oci://{ARTIFACTORY_REPOSITORY_URL}")
+    except subprocess.CalledProcessErro as e:
```

Comment:
```suggestion
    except subprocess.CalledProcessError as e:
```

Reply:
Should I just keep it like `except Exception as e:`

Reply:
subprocess.CalledProcessError is fine. that is the error raised when subprocess.run() is called and check=True. - https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError

> If check is true, and the process exits with a non-zero exit code, a [CalledProcessError](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError) exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured. ld the arguments, the exit code, and stdout and stderr if they were captured. )